provider/ec2/environ.go

```diff
@@ -425,18 +422,21 @@ func (e *environ) StartInstance(args environs.StartInstanceParams) (_ *environs.
 	// attached to the machine must all match, and must be the same
 	// as specified zone (if any).
 	availabilityZone, placementSubnetID, err := e.deriveAvailabilityZoneAndSubnetID(args)
-	switch {
-	case err != nil && errors.IsNotValid(err):
-		return nil, errors.Wrap(err, environs.ErrAvailabilityZoneFailed)
-	case err != nil:
-		return nil, err
+	if err != nil {
+		return nil, &common.StartInstanceError{
+			Err:             errors.Trace(err),
+			ZoneIndependent: !errors.IsNotValid(err),
```

Comment:
I find these hard to read. It *is* zone independent if it is not Not Valid?

Maybe a comment that explains the boolean logic in words?

Reply:
changed to

```go
zoneSpecific := errors.IsNotValid(err)
...
ZoneIndependent: !zoneSpecific,
```

and also added a comment